test/print_graph_tests.cc

```diff
@@ -84,22 +90,23 @@ TEST_CASE("command graph printing is unchanged", "[print_graph][command-graph]")
 	    "[1,1,1]}<br/><i>read_write</i> B0 {[0,0,0] - [1,1,1]}<br/><i>write</i> B0 {[0,0,0] - [1,1,1]}> fontcolor=black "
 	    "shape=box];}id_0_0->id_0_1[color=orchid];id_0_3->id_0_2[];id_0_5->id_0_2[color=limegreen];id_0_6->id_0_2[color=limegreen];id_0_7->id_0_2[color="
 	    "limegreen];id_0_3[label=<C3 on N0<br/><b>reduction</b> R1<br/> B0 {[0,0,0] - [1,1,1]}> fontcolor=black "
-	    "shape=ellipse];id_0_1->id_0_3[];id_0_4->id_0_3[];id_0_4[label=<C4 on N0<br/>(R1) <b>await push</b> transfer 8589934592 <br/>BB0 {[0,0,0] - "
-	    "[1,1,1]}> fontcolor=black shape=ellipse];id_0_0->id_0_4[color=orchid];id_0_5[label=<C5 on N0<br/>(R1) <b>push</b> transfer 8589934593 to N1<br/>BB0 "
-	    "[0,0,0] + [1,1,1]> fontcolor=black shape=ellipse];id_0_1->id_0_5[];id_0_6[label=<C6 on N0<br/>(R1) <b>push</b> transfer 8589934594 to N2<br/>BB0 "
-	    "[0,0,0] + [1,1,1]> fontcolor=black shape=ellipse];id_0_1->id_0_6[];id_0_7[label=<C7 on N0<br/>(R1) <b>push</b> transfer 8589934595 to N3<br/>BB0 "
-	    "[0,0,0] + [1,1,1]> fontcolor=black shape=ellipse];id_0_1->id_0_7[];}";
+	    "shape=ellipse];id_0_1->id_0_3[];id_0_4->id_0_3[];id_0_4[label=<C4 on N0<br/>(R1) <b>await push</b> transfer 8589934592 <br/>BB0 {[0,0,0] - [1,1,1]}> "
+	    "fontcolor=black shape=ellipse];id_0_0->id_0_4[color=orchid];id_0_5[label=<C5 on N0<br/>(R1) <b>push</b> transfer 8589934593 to N1<br/>BB0 [0,0,0] + "
+	    "[1,1,1]> fontcolor=black shape=ellipse];id_0_1->id_0_5[];id_0_6[label=<C6 on N0<br/>(R1) <b>push</b> transfer 8589934594 to N2<br/>BB0 [0,0,0] + "
+	    "[1,1,1]> fontcolor=black shape=ellipse];id_0_1->id_0_6[];id_0_7[label=<C7 on N0<br/>(R1) <b>push</b> transfer 8589934595 to N3<br/>BB0 [0,0,0] + "
+	    "[1,1,1]> fontcolor=black shape=ellipse];id_0_1->id_0_7[];}";
 
 	// fully check node 0
 	const auto dot0 = dctx.print_command_graph(0);
 	CHECK(dot0 == expected);
 
 	// only check the rough string length and occurence count of N1/N2... for other nodes
 	const int expected_occurences = count_occurences(expected, "N0");
-	for(size_t i = 1; i < num_nodes; ++i) {
-		const auto dot_n = dctx.print_command_graph(i);
-		REQUIRE_THAT(dot_n.size(), Catch::Matchers::WithinAbs(expected.size(), 50));
-		CHECK(count_occurences(dot_n, fmt::format("N{}", i)) == expected_occurences);
+	for(node_id nid = 1; nid < num_nodes; ++nid) {
+		CAPTURE(nid);
+		const auto dot_n = dctx.print_command_graph(nid);
+		CHECK_THAT(dot_n.size(), Catch::Matchers::WithinAbs(expected.size(), 200));
```

Comment:
Because after the fix, only chunk 0 read-writes and writes the buffer (which has size 1 and has a one_to_one RM). I have changed the example code somewhat to generate more comparable graphs on all nodes.